airflow/serialization/serde.py

```diff
@@ -270,38 +273,94 @@ def _stringify(classname: str, version: int, value: T | None) -> str:
-        for s in getattr(name, "serializers", list()):
-            if not isinstance(s, str):
-                s = qualname(s)
-            if s in _serializers and _serializers[s] != name:
-                raise AttributeError(f"duplicate {s} for serialization in {name} and {_serializers[s]}")
-            log.debug("registering %s for serialization")
-            _serializers[s] = name
-        for d in getattr(name, "deserializers", list()):
-            if not isinstance(d, str):
-                d = qualname(d)
-            if d in _deserializers and _deserializers[d] != name:
-                raise AttributeError(f"duplicate {d} for deserialization in {name} and {_serializers[d]}")
-            log.debug("registering %s for deserialization", d)
-            _deserializers[d] = name
-            _extra_allowed.add(d)
-
-
-def _compile_patterns():
-    patterns = conf.get("core", "allowed_deserialization_classes").split()
-
-    _patterns.clear()  # ensure to reinit
-    for p in patterns:
-        p = re.sub(r"(\w)\.", r"\1\..", p)
-        _patterns.append(re.compile(p))
-
-
-_register()
-_compile_patterns()
+_AIRFLOW_SERIALIZERS = {
+    "decimal": {"Decimal": "airflow.serialization.serializers.bignum"},
+    "datetime": {
+        "date": "airflow.serialization.serializers.datetime",
+        "datetime": "airflow.serialization.serializers.datetime",
+        "timedelta": "airflow.serialization.serializers.datetime",
+    },
+    "kubernetes": {
+        "client.models.v1_resource_requirements.V1ResourceRequirements": (
+            "airflow.serialization.serializers.kubernetes"
+        ),
+        "client.models.v1_pod.V1Pod": "airflow.serialization.serializers.kubernetes",
+    },
+    "numpy": {
+        "bool_": "airflow.serialization.serializers.numpy",
+        "complex64": "airflow.serialization.serializers.numpy",
+        "complex128": "airflow.serialization.serializers.numpy",
+        "float16": "airflow.serialization.serializers.numpy",
+        "float64": "airflow.serialization.serializers.numpy",
+        "intc": "airflow.serialization.serializers.numpy",
+        "intp": "airflow.serialization.serializers.numpy",
+        "int8": "airflow.serialization.serializers.numpy",
+        "int16": "airflow.serialization.serializers.numpy",
+        "int32": "airflow.serialization.serializers.numpy",
+        "int64": "airflow.serialization.serializers.numpy",
+        "uint8": "airflow.serialization.serializers.numpy",
+        "uint16": "airflow.serialization.serializers.numpy",
+        "uint32": "airflow.serialization.serializers.numpy",
+        "uint64": "airflow.serialization.serializers.numpy",
+    },
+    "pendulum": {
+        "datetime.DateTime": "airflow.serialization.serializers.datetime",
+        "tz.timezone.FixedTimezone": "airflow.serialization.serializers.timezone",
+        "tz.timezone.Timezone": "airflow.serialization.serializers.timezone",
+    },
+}
+
+_AIRFLOW_DESERIALIZERS = {
```

Comment:
We don't have a deserializer for Kube?

Reply:
Yes apparently, this is the case before the patch and I’m not sure why.

Reply:
cc @bolkedebruin

Reply:
It isn't there as it is only used within DAG serialization which I havent implemented yet. But please see below, I think this 'sync' issue should be prevented by implementing the change differently.